pkg/controller/batchrelease/context/context.go

```diff
@@ -64,6 +64,8 @@ type BatchContext struct {
 	// the next two fields are only used for bluegreen style
 	CurrentSurge intstr.IntOrString `json:"currentSurge,omitempty"`
 	DesiredSurge intstr.IntOrString `json:"desiredSurge,omitempty"`
+	// batches field is added to support complicated scenes such as batch-id label
+	Batches []v1beta1.ReleaseBatch
```

Comment:
can we change the LabelPatcher interface, and add ReleaseBatch in the PatchPodBatchLabel func.

Reply:
done